sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/encrypt_parameters.hpp

```diff
@@ -0,0 +1,284 @@
+     * and returned in the #EncryptResult after encryption.
+     *
+     * @param plaintext The plaintext to encrypt.
+     * @param additionalAuthenticationData Optional data that is authenticated but not encrypted.
+     * @return An instance of the #EncryptParameters class for the #EncryptionAlgorithm::A128Gcm
+     * encryption algorithm.
+     */
+    static EncryptParameters A128GcmParameters(
+        std::vector<uint8_t> const& plaintext,
+        std::vector<uint8_t> const& additionalAuthenticationData)
+    {
+      return EncryptParameters(
+          EncryptionAlgorithm::A128Gcm, plaintext, {}, additionalAuthenticationData);
+    }
+
+    /**
+     * @brief Creates an instance of the #EncryptParameters class for the
+     * #EncryptionAlgorithm::A192Gcm encryption algorithm. The nonce will be generated automatically
+     * and returned in the #EncryptResult after encryption.
+     *
+     * @param plaintext The plaintext to encrypt.
+     * @param additionalAuthenticationData Optional data that is authenticated but not encrypted.
+     * @return An instance of the #EncryptParameters class for the #EncryptionAlgorithm::A192Gcm
+     * encryption algorithm.
+     */
+    static EncryptParameters A192GcmParameters(
+        std::vector<uint8_t> const& plaintext,
+        std::vector<uint8_t> const& additionalAuthenticationData)
+    {
+      return EncryptParameters(
+          EncryptionAlgorithm::A192Gcm, plaintext, {}, additionalAuthenticationData);
+    }
+
+    /**
+     * @brief Creates an instance of the #EncryptParameters class for the
+     * #EncryptionAlgorithm::A256Gcm encryption algorithm. The nonce will be generated automatically
+     * and returned in the #EncryptResult after encryption.
+     *
+     * @param plaintext The plaintext to encrypt.
+     * @param additionalAuthenticationData Optional data that is authenticated but not encrypted.
+     * @return An instance of the #EncryptParameters class for the #EncryptionAlgorithm::A256Gcm
+     * encryption algorithm.
+     */
+    static EncryptParameters A256GcmParameters(
+        std::vector<uint8_t> const& plaintext,
+        std::vector<uint8_t> const& additionalAuthenticationData)
+    {
+      return EncryptParameters(
+          EncryptionAlgorithm::A256Gcm, plaintext, {}, additionalAuthenticationData);
+    }
+
+    /**
+     * @brief Creates an instance of the #EncryptParameters class for the
+     * #EncryptionAlgorithm::A128Cbc encryption algorithm.
+     *
+     * @param plaintext The plaintext to encrypt.
+     * @param iv Optional initialization vector. If null, a cryptographically random initialization
+     * vector will be generated using #RandomNumberGenerator.
+     * @return An instance of the #EncryptParameters class for the #EncryptionAlgorithm::A128Cbc
+     * encryption algorithm.
+     */
+    static EncryptParameters A128CbcParameters(
+        std::vector<uint8_t> const& plaintext,
+        std::vector<uint8_t> const& iv)
```

Comment:
iv is optional and should be cryptographically generated, unless you're leaving that for the caller. Given the description, it seems not - thus, shouldn't this have a default value?